doc/ale.txt

```diff
@@ -1421,18 +1421,35 @@ ale#linter#Define(filetype, linter)                       *ale#linter#Define()*
                          This argument may be omitted or `''` when a linter
                          does not represent an LSP linter.
 
+                         When this argument is set to `'socket'`, then the
+                         linter will be defined as an LSP linter which
+                         communicates with a running language server at
+                         an address provided by `address_callback`.
+                         With `socket`, none of `executable`,
+                         `executable_callback`, `command`, `command_callback`,
+                         `command_chain` may be provided.
```

Comment:
I think let's hold off on documenting this for now, as it isn't yet fully implemented. I only want to document something if  it's tested and proven to work to some degree.